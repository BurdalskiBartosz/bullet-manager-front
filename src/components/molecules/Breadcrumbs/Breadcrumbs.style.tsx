import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const Wrapper = styled.div``;

export const BreadLink = styled(Link)`
	text-decoration: none;
	margin-right: 1rem;
	font-size: 2.5rem;
`;
