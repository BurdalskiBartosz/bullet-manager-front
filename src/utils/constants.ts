export const baseURL: string = 'http://localhost:3001/api';

export const appRootRoute: string = 'app';
