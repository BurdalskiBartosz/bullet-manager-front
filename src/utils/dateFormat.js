import { format } from 'date-fns';

export const dateFormat = (value) => format(value, 'MM/dd/yyyy');
