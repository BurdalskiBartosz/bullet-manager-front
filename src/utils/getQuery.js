export const getQuery = (object) => {
	if (!object) return '';
	const queryStrings = Object.entries(object).map(([key, value]) => {
		const stringKey = encodeURIComponent(key);
		if (typeof value === 'object')
			return `${stringKey}=${JSON.stringify(value)}`;
		else return `${stringKey}=${encodeURIComponent(value)}`;
	});

	if (queryStrings.length === 0) return '';
	return '?' + queryStrings.join('&');
};
