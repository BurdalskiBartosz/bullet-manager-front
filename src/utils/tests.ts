import { render } from '@testing-library/react';
import Providers from '../providers';

const customRender = (ui: any) => render(ui, { wrapper: Providers });

export * from '@testing-library/react';

export { customRender as render };
