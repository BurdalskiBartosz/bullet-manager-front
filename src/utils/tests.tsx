import { render, RenderOptions } from '@testing-library/react';
import { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { AppStore, RootState, setupStore } from 'store/store';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { BrowserRouter } from 'react-router-dom';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}
const reduxStore = setupStore();
export const renderWithProviders = (
	ui: React.ReactElement,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = reduxStore,
		...renderOptions
	}: ExtendedRenderOptions = {}
) => {
	const Wrapper = ({ children }: PropsWithChildren<{}>) => {
		return (
			<BrowserRouter>
				<Provider store={store}>
					<ThemeProvider theme={theme}>{children}</ThemeProvider>
				</Provider>
			</BrowserRouter>
		);
	};

	// Return an object with the store and all of RTL's query functions
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders as render };
export * from '@testing-library/react';
