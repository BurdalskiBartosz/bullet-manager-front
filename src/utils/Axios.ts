/* eslint-disable no-param-reassign */
import axios, { AxiosRequestConfig, AxiosError, AxiosResponse } from 'axios';
import { baseURL } from './constants';

const options: AxiosRequestConfig = {
	baseURL,
	timeout: 10000,
	headers: { 'Content-Type': 'application/json' },
	withCredentials: true
};

export const fetchProvider = axios.create(options);
const { interceptors } = fetchProvider;

const getRequestConfig = (config: AxiosRequestConfig) => {
	const token: string | null = window.localStorage.getItem('token');

	if (token) config.headers!.Authorization = `Bearer ${token}`;

	return config;
};

const getRequestError = (error: AxiosError) => Promise.reject(error);

const getResponse = (response: AxiosResponse) => response;

const getResponseError = (error: AxiosError) => {
	const errorResponse =
		error.response?.status === 400 ? error : error.response;
	return Promise.reject(errorResponse);
};

interceptors.request.use(getRequestConfig, getRequestError);
interceptors.response.use(getResponse, getResponseError);

export const instanceForMocks = axios.create(options);
