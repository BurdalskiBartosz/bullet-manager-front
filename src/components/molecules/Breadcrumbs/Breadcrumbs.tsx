import { FC } from 'react';
import { useLocation } from 'react-router-dom';
import { appRootRoute } from 'utils/constants';
import { Font, Icon } from 'components';
import { BreadLink, Wrapper } from './Breadcrumbs.style';

type tProps = {};

const Breadcrumbs: FC<tProps> = () => {
	const { pathname } = useLocation();
	const getPaths = () => {
		const pathnames = pathname
			.split('/')
			.filter((el) => el && el !== appRootRoute)
			.map((el, i, arr) => {
				const link = arr
					.slice(0, i + 1)
					.reduce((prev, curr) => `${prev}/${curr}`, '');
				const routeName = el[0].toUpperCase() + el.slice(1);
				const isLast = arr.length > i + 1 ? true : false;
				return (
					<BreadLink
						key={routeName}
						to={`/${appRootRoute}${link}`}
						style={{ color: 'white' }}
					>
						<Font style={{ marginRight: '1rem' }}>{routeName}</Font>
						{isLast && (
							<Icon
								fill="#FFF"
								iconName="arrow_right"
								height="15"
							/>
						)}
					</BreadLink>
				);
			});
		return pathnames;
	};
	return <Wrapper data-testid="breadcrumbs-component">{getPaths()}</Wrapper>;
};

export default Breadcrumbs;
