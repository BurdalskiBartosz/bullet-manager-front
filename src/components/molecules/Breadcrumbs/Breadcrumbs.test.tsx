import { render, screen } from 'utils/tests';
import Breadcrumbs from './Breadcrumbs';

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useLocation: () => ({
		pathname: '/app/dashboard/example/path'
	})
}));

describe('Breadcrumbs', () => {
	it('should correctly', () => {
		render(<Breadcrumbs />);
		const breadcrumbs = screen.getByTestId(/breadcrumbs-component/i);
		expect(breadcrumbs).toMatchSnapshot();
	});
});
