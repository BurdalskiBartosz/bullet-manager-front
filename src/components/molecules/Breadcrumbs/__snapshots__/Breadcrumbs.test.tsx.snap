// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should correctly 1`] = `
.c2 {
  margin-right: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-right: 1rem;
  font-size: 2.5rem;
}

<div
  class="c0"
  data-testid="breadcrumbs-component"
>
  <a
    class="c1"
    href="/app/dashboard"
    style="color: white;"
  >
    <p
      class="c2"
    >
      Dashboard
    </p>
    <svg
      data-testid="icon-component"
      fill="#FFF"
      height="15"
      width="20px"
    >
      arrow_right.svg
    </svg>
  </a>
  <a
    class="c1"
    href="/app/dashboard/example"
    style="color: white;"
  >
    <p
      class="c2"
    >
      Example
    </p>
    <svg
      data-testid="icon-component"
      fill="#FFF"
      height="15"
      width="20px"
    >
      arrow_right.svg
    </svg>
  </a>
  <a
    class="c1"
    href="/app/dashboard/example/path"
    style="color: white;"
  >
    <p
      class="c2"
    >
      Path
    </p>
  </a>
</div>
`;
